ocaml/parsing/jane_syntax.ml

```diff
@@ -1930,6 +1939,10 @@ module Expression = struct
     | Language_extension Labeled_tuples ->
       let expr, attrs = Labeled_tuples.of_expr expr in
       Some (Jexp_tuple expr, attrs)
+    | Language_extension Mode -> (
+      match Modes.of_expr expr with
+      | Some (expr, attrs) -> Some (Jexp_modes expr, attrs)
+      | None -> None)
```

Comment:
Is it correct/better to `Misc.fatal_error` in the `None` case?

Reply:
Thanks Nick - if it ends up taking too much time, let's just leave it as it is.

Reply:
The latest commit allow us to use `Misc.fatal_error` here. Resolve this once you convince yourself it is correct.